File: Backend/AssignmentManagementSystem.BusinessLogicLayer/Entities/Subject.cs
using System;
using System.Collections.Generic;

namespace AssignmentManagementSystem.BusinessLogicLayer.Entities;

public partial class Subject
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public string? Code { get; set; }
    public Guid ClassId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public virtual ICollection<Assignment> Assignments { get; set; } = new List<Assignment>();
    public virtual Class Class { get; set; } = null!;
    public virtual ICollection<TeacherSubject> TeacherSubjects { get; set; } = new List<TeacherSubject>();
}

File: Frontend/src/types/subject.ts
export interface Subject {
    id: string;
    name: string;
    code: string | null;
    classId: string;
    className?: string;
    createdAt?: string;
    updatedAt?: string;
}

export interface CreateSubjectData {
    name: string;
    code: string | null;
    classId: string;
}

export interface UpdateSubjectData {
    name: string;
    code: string | null;
    classId: string;
}